src/code/templates/egg-game.js

```diff
@@ -101,44 +185,375 @@ var animationEvents = {
+        addChromAnimStage(BioLogica.MALE, defaultAnimationSpeed, '2', ['a', 'b']);
+        addChromAnimStage(BioLogica.MALE, defaultAnimationSpeed, 'XY', ['x', 'y']);
+        addGameteAnimStage(BioLogica.MALE, defaultAnimationSpeed);
+
+        if (debugTotalGameteCount !== 1) {
+          // randomize mother's chromosomes simultaneously
+          addGenomeAnimStage(BioLogica.FEMALE, enhancedAnimationSpeed);
+          addGameteAnimStage(BioLogica.FEMALE, enhancedAnimationSpeed);
+
+          // randomize father's chromosomes simultaneously
+          addGenomeAnimStage(BioLogica.MALE, enhancedAnimationSpeed);
+          addGameteAnimStage(BioLogica.MALE, enhancedAnimationSpeed);
+        }
+
+        // randomize the remaining gametes for both parents simultaneously
+        const totalGameteCount = debugTotalGameteCount || 8,
+              fullCycleCount = totalGameteCount - 2;
+        for (let i = 0; i < fullCycleCount; ++i) {
+          addGenomeAnimStage(BOTH_SEXES, enhancedAnimationSpeed);
+          addGameteAnimStage(BOTH_SEXES, enhancedAnimationSpeed);
+        }
+
+        stages.push({ type: 'complete' });
+      }
+
+      function selectStageChromosomes(speed, chroms) {
+        if (!_this) return;
+        let animatingGametes = cloneDeep(_this.state.animatingGametes) || [{}, {}, {}, {}];
+        chroms.forEach(({ sex, name }) => {
+          let gamete = animatingGametes && animatingGametes[sex],
+              side = gamete && gamete[name];
+          if (side) {
+            const parentId = sex === BioLogica.FEMALE ? 'mother' : 'father',
+                  chromContainerId = `${parentId}${name}${side}`,
+                  elt = document.getElementById(chromContainerId).parentNode;
+            // this will trigger an animation which will call onFinish() when complete,
+            // which will trigger the next stage of the animation.
+            _this.selectChromosomes(sex, speed, [{name, side, elt}],
+                                    animationEvents.randomizeChromosomes.onFinish);
+          }
+        });
+        animatingGametes[2] = clone(animatingGametes[0]);
+        animatingGametes[3] = clone(animatingGametes[1]);
+        // delay appearance of labels until the chromosomes arrive
+        setTimeout(() => _this.setState({ animatingGametes }), delayShowSelectedChromosomeLabels);
+      }
+
+      function toggleAnimatingGameteChromosome(sex, chromName, sides) {
+        let animatingGametes = _this.state.animatingGametes || [{}, {}, {}, {}],
+            gamete = animatingGametes && animatingGametes[sex],
+            currSide = gamete && gamete[chromName],
+            newSide = currSide != null
+                        ? (currSide === sides[0] ? sides[1] : sides[0])
+                        : oneOf(sides);
+        animatingGametes[sex] = assign({}, gamete, { [chromName]: newSide });
+        if (_this) _this.setState({ animatingGametes });
```

Comment:
I think your approach is a good one. I was/am familiar with the discussion around isMounted being an anti-pattern, but also think that discussion glosses over the fact that many asynchronous actions can't easily be interrupted. In our case, however, we certainly have the ability to interrupt our timer tasks and I would expect react-motion to be handling the lifecycle events appropriately for its own animations. (I think I was unconvinced of that at the time I was adding the `if (_this)` tests.) If you have some working code then adding a commit to the PR would be great. Thanks